packages/opencensus-propagation-b3/src/b3Format.ts

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {types} from '@opencensus/opencensus-core';
+import {classes} from '@opencensus/opencensus-core';
+import {logger} from '@opencensus/opencensus-core';
+import * as crypto from 'crypto';
+import * as uuid from 'uuid';
+
+import {Tracer} from '../../opencensus-core/build/src/index-classes';
+
+const X_B3_TRACE_ID = 'x-b3-traceid';
+const X_B3_SPAN_ID = 'x-b3-spanid';
+const X_B3_PARENT_SPAN_ID = 'x-x3-parentspanid';
+const X_B3_SAMPLED = 'x-b3-sampled';
+
+const SAMPLED_VALUE = 0x1;
+const NOT_SAMPLED_VALUE = 0x0;
+
+
+/** Propagates span context through B3 Format propagation. */
+export class B3Format implements types.Propagation {
+  /**
+   * Gets the trace context from a request headers. If there is no trace context
+   * in the headers, null is returned.
+   * @param getter
+   */
+
+  extract(getter: types.HeaderGetter): types.SpanContext {
+    // tslint:disable-next-line
+    if (getter) {
+      const opt = getter.getHeader(X_B3_SAMPLED);
```

Comment:
Question: are this logic valid if any of the `getHeader` operations returns a `string[]`.